apidump/apidump.go

```diff
@@ -369,6 +370,12 @@ func Run(args Args) error {
 		}
 	}
 
+	// Create a buffer pool for storing HTTP payloads.
+	pool, err := buffer_pool.MakeBufferPool(20*1024*1024, 4*1024)
```

Comment:
I would like to see some back-of-the-envelope estimate on how much this limits our collection rate.

Flickr is 380,000 events per day = 260 per second.  Little's law says that if they spend 40ms resident then that is 10.4 requests resident a time, call it 12.  That is a generous 1.8MB per request.

But, we actually throw a lot of Flickr's requests away after parsing them!  Their true data rate may be much higher; we could look at the telemetry to figure out how much... if they were successfully sending telemetry, dangit.

If we're throwing away 90% of their requests, that changes the allocation to 200KB per request.  Still seems OK.

Reply:
The other thing to consider is how much this increases our fixed overhead, and whether we should reduce the GC parameter to compensate.

20MB of live data = 40MB of RSS footprint

If our footprint used to be 400MB, it's now 440MB.  But we could get that back by setting GOGCLIMIT=81 instead.  (180MB + 220MB live heap = 400MB, 220 * 0.81 =~ 180